pandas/core/apply.py

```diff
@@ -123,15 +123,20 @@ def __init__(
         self.result_type = result_type
 
         # curry if needed
-        if (
-            (kwargs or args)
-            and not isinstance(func, (np.ufunc, str))
-            and not is_list_like(func)
-        ):
-
-            def f(x):
-                return func(x, *args, **kwargs)
-
+        if (kwargs or args) and not isinstance(func, (np.ufunc, str)):
+            if not is_list_like(func):
+
+                def f(x):
+                    return func(x, *args, **kwargs)
+
+            elif type(func) == list and all(i for i in func if callable(i)):
+                # GH 50624
+                # only explicit arg passing is supported temporarily
+                # eg. df.agg([foo], a=1)
+                # df.agg([foo], 1) is not supported
+                f = _recreate_func(func, kwargs)
```

Comment:
Is there a reason why passing `*self.args` and `**self.kwargs` on L335 and L347 in this file won't work?

Reply:
To be honest, I limited my thinking and didn't notice L335 to L347.

I'm going to take some time to reorganize it. Thanks for the reminder!